### DISCUSSION TITLE
Get a font's kerning data

### DISCUSSION THREAD

**Discussion by ThomasMiz** [SixLabors/Fonts#320] [General] [answered]
I am upgrading a library that used a previous beta version of SixLabors.Fonts, in which we could get the kerning from `IFontInstance.GetOffset()`, however I can't find any of this functionality on the new API. On the contrary, after looking around all the kerning functionality I could find on the library's code appears to be internal.

How can I retrieve data from a font's kerning table? I am currently on 1.0.0-beta19.

**Comment by JimBobSquarePants** [comment]
What version did you upgrade from? 

From memory that method was removed because it did not tell the complete story. A glyph will be offset by a combination of glyph shaping and kerning.

**Reply by ThomasMiz** [reply]
Hi Jim! Thanks for the help

I am upgrading from version 1.0.0-beta0013

**Reply by JimBobSquarePants** [reply]
Ah yes. That's a while back. Since that release we've implemented [text shaping](https://github.com/SixLabors/Fonts/pull/198) so that method didn't return a result that would indicate the whole story. What's your use case?

OK... So I can see this. It looks like you're building your own scaled font metrics type.  
https://github.com/SilkCommunity/TrippyGL/pull/5/files#diff-b6ed35d890d8e584992c28c5148cdc861cdd3ed3ec8cb335af1c174394515265R207

Shouldn't be a biggie to re-add it.

**Reply by ThomasMiz** [reply]
Yep! I'm basically rasterizing a font in a specific size to create a file that contains a png (the "atlas"), and tables for advances (one per glyph) and kerning (one per glyph pair).

So far I've got the exact same atlas image generated. I can't get the metrics to match though, the old code could generate an atlas that produced text that I would overlay with same-sized text from Paint.NET and you could see that the kerning and advances were exactly identical. Now I'm trying to figure out how to get the calculations right to get the same results, and how to get the kerning.

You even looked up the project! Thanks a lot for the help :-)
You reckon one of the maintainers could re-add it? I haven't touched this codebase before but if you think it's an easy change then maybe I can give it a shot

**Reply by JimBobSquarePants** [reply, marked as answer]
No worries 😉 

I can get that re-added with little effort. 

What other metric discrepancies are you seeing? I’ve reworked the metrics over time to match FreeType output. I can normally overlay the output over browser generated equivalents without discernible discrepancy.
